remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -101,7 +103,8 @@ public void addItem(RemixerItem remixerItem) {
     }
     for (RemixerItem remove : itemsToRemove) {
       listForKey.remove(remove);
-      remixerItems.remove(remove);
+      // no need to remove from contextMap since that will be cleared just by the context being
+      // reclaimed, that's why it's a WeakHashMap.
```

Comment:
part of me still thinks you should keep this explicit removal. What happens in this case:
```
contextMap = {
  MainActivity : [ foo_var, bar_var ]
}
```

And `addItem()` causes `foo_var` to be removed. If you do not have explicit removal, won't `foo_var` stick around? It's not going to cause a memory leak or anything, but isn't it more clean to remove it explicitly?